### PR TITLE
Make sure to exit with 0 of 1 when reloading the framework

### PR DESCRIPTION
**Pull request by bucaran**
Make sure to exit with 0 of 1 when reloading the framework.

This paves the way for [this](https://github.com/bpinto/oh-my-fish/issues/307#issuecomment-69794795) too.

### CODE REVIEW

**Comment by bpinto** [comment]
I don't understand this, `set fish_function_path $user_function_path $fish_function_path` should return always true, right?

P.S.: Can we add a test for this?

**Comment by bucaran** [comment]
This is necessary to allow halting individual test execution. What happens is simply that the 

```fish
if not contains $fish_path/functions/ $fish_function_path
  set fish_function_path $fish_path/functions/ $fish_function_path
end
```
When you reload the session, $fish_path is already added to $fish_function_path,  so the condition modifies the `$status` variable causing the reload to exit with 1.

Try using a theme that shows a visual cue like changing the color of the prompt after a command runs like _bobthefish_ and try reloading `. $fish_path/oh-my-fish.fish` and you will see.

**Comment by bpinto** [comment]
I have my head wrapped on something else for now, but adding a test case would be nice.

**Comment by bucaran** [comment]
What kind of test?

**Comment by bpinto** [comment]
That running `. $fish_path/oh-my-fish.fish` returns 0.

**Comment by bucaran** [comment]
Ahh, sure, that should be simple to add.

**Comment by bucaran** [comment]
Done.

**Comment by bpinto** [comment]
Does the test fail if you remove your code?

**Comment by bucaran** [comment]
Testing....
__Yes__.

yes

Merge order should be:
+ reload-and-exit-ok
+ stop-test-continue-suite

**Comment by bpinto** [comment]
Awesome! If we add a test for every bug we found, this will be amazing!

My goal is to start adding test for the `fish-spec` plugin, it's a little bit harder, but required.

File: spec/oh-my-fish.spec.fish [hunk 1]
```diff
@@ -69,6 +69,13 @@ function describe_oh_my_fish -d "Oh-My-Fish test spec..."
     load_oh_my_fish
     expect $fish_function_path --to-contain $fish_path/themes/l
   end
+
+  function it_reloads_with_status_of_0 \
+    -d "It should reload the framework with $status of 0"
```
Comment (bpinto) [inline spec/oh-my-fish.spec.fish]: Shouldn't this $status be escaped?

**Comment by bucaran** [comment]
> My goal is to start adding test for the fish-spec plugin, it's a little bit harder, but required.

Absolutely.

File: spec/oh-my-fish.spec.fish (hunk 1 above)
Comment (bucaran) [inline spec/oh-my-fish.spec.fish]: Yes. Done.

Comment (bpinto) [inline spec/oh-my-fish.spec.fish]: Where? :P

Comment (bucaran) [inline spec/oh-my-fish.spec.fish]: Sorry, I forgot to `-f` force the push since I squashed the \$ bug.

**Comment by bpinto** [comment]
@bucaran You are still using "should", I'll convince you. :)

**Comment by bucaran** [comment]
You mean for the long descriptions? Where where? :laughing:

**Comment by bpinto** [comment]
:P

**Pull request merged**